relayer/processor/path_processor_internal.go

```diff
@@ -869,29 +932,59 @@ func (pp *PathProcessor) connectionMessagesToSend(pathEnd1ConnectionHandshakeRes
+func (pp *PathProcessor) packetMessagesToSend(
+	channelPairs []channelPair,
+	pathEnd1ProcessRes []pathEndPacketFlowResponse,
+	pathEnd2ProcessRes []pathEndPacketFlowResponse,
+) ([]packetIBCMessage, []packetIBCMessage, []channelIBCMessage, []channelIBCMessage) {
 	pathEnd1PacketLen := 0
 	pathEnd2PacketLen := 0
+	pathEnd1ChannelLen := 0
+	pathEnd2ChannelLen := 0
+
 	for i := 0; i < len(channelPairs); i++ {
 		pathEnd1PacketLen += len(pathEnd2ProcessRes[i].DstMessages) + len(pathEnd1ProcessRes[i].SrcMessages)
 		pathEnd2PacketLen += len(pathEnd1ProcessRes[i].DstMessages) + len(pathEnd2ProcessRes[i].SrcMessages)
+		pathEnd1ChannelLen += len(pathEnd2ProcessRes[i].DstChannelMessage)
+		pathEnd2ChannelLen += len(pathEnd1ProcessRes[i].DstChannelMessage)
 	}
 
 	pathEnd1PacketMessages := make([]packetIBCMessage, 0, pathEnd1PacketLen)
 	pathEnd2PacketMessages := make([]packetIBCMessage, 0, pathEnd2PacketLen)
 
+	pathEnd1ChannelMessage := make([]channelIBCMessage, 0, pathEnd1ChannelLen)
+	pathEnd2ChannelMessage := make([]channelIBCMessage, 0, pathEnd2ChannelLen)
+
 	for i, channelPair := range channelPairs {
 		pathEnd1PacketMessages = append(pathEnd1PacketMessages, pathEnd2ProcessRes[i].DstMessages...)
 		pathEnd1PacketMessages = append(pathEnd1PacketMessages, pathEnd1ProcessRes[i].SrcMessages...)
 
 		pathEnd2PacketMessages = append(pathEnd2PacketMessages, pathEnd1ProcessRes[i].DstMessages...)
 		pathEnd2PacketMessages = append(pathEnd2PacketMessages, pathEnd2ProcessRes[i].SrcMessages...)
 
+		pathEnd1ChannelMessage = append(pathEnd1ChannelMessage, pathEnd2ProcessRes[i].DstChannelMessage...)
+		pathEnd2ChannelMessage = append(pathEnd2ChannelMessage, pathEnd1ProcessRes[i].DstChannelMessage...)
+
+		pp.pathEnd1.messageCache.ChannelHandshake.DeleteMessages(pathEnd2ProcessRes[i].ToDeleteDstChannel)
+		pp.pathEnd2.messageCache.ChannelHandshake.DeleteMessages(pathEnd1ProcessRes[i].ToDeleteDstChannel)
+
 		pp.pathEnd1.messageCache.PacketFlow[channelPair.pathEnd1ChannelKey].DeleteMessages(pathEnd1ProcessRes[i].ToDeleteSrc, pathEnd2ProcessRes[i].ToDeleteDst)
 		pp.pathEnd2.messageCache.PacketFlow[channelPair.pathEnd2ChannelKey].DeleteMessages(pathEnd2ProcessRes[i].ToDeleteSrc, pathEnd1ProcessRes[i].ToDeleteDst)
 		pp.pathEnd1.packetProcessing[channelPair.pathEnd1ChannelKey].deleteMessages(pathEnd1ProcessRes[i].ToDeleteSrc, pathEnd2ProcessRes[i].ToDeleteDst)
 		pp.pathEnd2.packetProcessing[channelPair.pathEnd2ChannelKey].deleteMessages(pathEnd2ProcessRes[i].ToDeleteSrc, pathEnd1ProcessRes[i].ToDeleteDst)
 	}
 
-	return pathEnd1PacketMessages, pathEnd2PacketMessages
+	return pathEnd1PacketMessages, pathEnd2PacketMessages, pathEnd1ChannelMessage, pathEnd2ChannelMessage
+}
+
+// orderFromString parses a string into a channel order byte
+func orderFromString(order string) chantypes.Order {
+	switch strings.ToUpper(order) {
+	case "UNORDERED":
```

Comment:
`chantypes.UNORDERED.String()` here and `ORDERED`?

Also maybe this should go in `types_internal.go`?

Reply:
Good idea, I brought this in a bit haphazardly.